numpyro/handlers.py

```diff
@@ -245,6 +246,20 @@ def process_message(self, msg):
             msg['stop'] = True
 
 
+def _eager_expand_fn(fn):
```

Comment:
Do I understand correctly that this function eliminates lazy `ExpandedDistribution`s and replaces them with eagerly `tree_map`-expanded distributions, and that this is needed because Funsor distributions do not support Expanded distributions? If so could you add a commend explaining that?

I guess an alternative would be to support `ExpandedDistribution`s in Funsor, or to do this conversion in the funsor layer. @eb8680 does that seem possible?

Reply:
Yes, it is. I'll add a comment for clarity. I thought from our last discussion, this stuff should be treated in Pyro/NumPyro?

Reply:
> I guess an alternative would be to support ExpandedDistributions in Funsor, or to do this conversion in the funsor layer.

We could certainly implement an equivalent of `.expand`. I think the easiest thing to do to address this case would be to add a `to_funsor` pattern for `ExpandedDistribution`.

Reply:
> add a `to_funsor` pattern for `ExpandedDistribution`

Yeah, it seems that you both agree to do this. I can take an effort for it.

Reply:
I think it would basically be the same as what you have here, plus a final step that calls `to_funsor` on the eagerly expanded result.